### DISCUSSION TITLE
Intermittent error: An error occurred while fetching the data. Please try again.

### DISCUSSION THREAD

**Discussion by sagerock** [vercel/community#2319] [Help] [answered]
### Summary

I've got an application running here: https://gpt4-pdf-chatbot-langchain-for-alconox-icrpl3pye-sagerock.vercel.app/

It is using langchain and is calling data from Pinecone.

It always works when I run it locally. But when I use it at the Vercel domain it will run for a few times and then I get the error: "An error occurred while fetching the data. Please try again."

If I refresh the page I can run it again for a few times. And then the error comes back. 

Do you have any thoughts as to why this might be happening?

Thank you!

### Additional information

The error message is: "An error occurred while fetching the data. Please try again."

My logs say it is a 504 error code.

### Example

https://gpt4-pdf-chatbot-langchain-for-alconox-icrpl3pye-sagerock.vercel.app/

**Comment by amyegan** [comment, marked as answer]
Hi @sagerock! A `504` would be a [FUNCTION_INVOCATION_TIMEOUT](https://vercel.com/docs/errors/application-error-codes#function_invocation_timeout) or [EDGE_FUNCTION_INVOCATION_TIMEOUT](https://vercel.com/docs/errors/application-error-codes#edge_function_invocation_timeout), likely because the data connection is taking too long to return. 

You can find the serverless function execution timeout and edge function initial response timeout limits in our [limits & limitations page](https://vercel.com/docs/concepts/limits/overview#:~:text=%E2%88%9E-,Serverless%20Function%20Execution%20Timeout%20(Seconds),30,-Proxied%20Request%20Timeout). If you're using edge functions, then you may be able to use [streaming](https://vercel.com/docs/concepts/functions/edge-functions/streaming#) to solve the problem.

**Reply by sagerock** [reply]
This is so helpful. Thank you. I'll give this a try. Have a great day.
